internal/apim_factory.go

```diff
@@ -0,0 +1,7 @@
+package internal
+
+import "github.com/TykTechnologies/gateway-sdk/pkg/apim"
+
+type ApimFactory struct {
```

Comment:
I would suggest renaming `ApimFactory` to `ApimClient`. 

Factory is specific method pattern approach, which is not used here, as a result it makes code confusing.